src/Symfony/Component/DependencyInjection/Compiler/ServiceReferenceGraph.php

```diff
@@ -85,23 +85,26 @@ public function clear()
      * @param string $destValue
      * @param string $reference
      * @param bool   $lazy
+     * @param bool   $weak
      */
-    public function connect($sourceId, $sourceValue, $destId, $destValue = null, $reference = null/*, bool $lazy = false*/)
+    public function connect($sourceId, $sourceValue, $destId, $destValue = null, $reference = null/*, bool $lazy = false, bool $weak = false*/)
```

Comment:
should we tag this class as `@final` to allow us to avoid having to do such magic for future changes to the container ? I'm not sure extending the ServiceReferenceGraph actually makes sense.

Tagging it as `@internal` would be even more strict, but tools wanting to analyze the DIC to display debugging info may be a valid use case for *using* this class so I would not go that far (for instance, https://github.com/schmittjoh/JMSDebuggingBundle relies on it, even though I'm not sure it plays well with recent versions as I haven't used this bundle since years)

Reply:
I agree, `@final` it is now